src/main/java/jose/Duke.java

```diff
@@ -0,0 +1,126 @@
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Method containing the main logic of the program.
+     */
+    public void run() {
+        Scanner scanner = new Scanner(System.in);
+        Parser parser = new Parser();
+        boolean isModified = false;
+        boolean isRunning = true;
+
+        ui.showGreeting();
+
+        while (isRunning) {
+            String input = scanner.nextLine();
+            String[] taskInfo;
+            Task tempTask;
+
+            try {
+                Parser.Command command = parser.parse(input);
+                switch (command) {
+                case BYE:
+                    ui.showExitMessage();
+                    isRunning = false;
+                    break;
+                case LIST:
+                    ui.showList(tasks.getTasks());
+                    break;
+                case MARK:
+                    tempTask = tasks.getTask(parser.getIndex(input));
+                    tempTask.mark();
+                    isModified = true;
+                    ui.showMarkMessage(tempTask);
+                    break;
+                case UNMARK:
+                    tempTask = tasks.getTask(parser.getIndex(input));
+                    tempTask.unmark();
+                    isModified = true;
+                    ui.showUnmarkMessage(tempTask);
+                    break;
+                case DELETE:
+                    tempTask = tasks.getTask(parser.getIndex(input));
+                    ui.showDeleteMessage(tempTask);
+                    tasks.removeTask(tempTask);
+                    isModified = true;
+                    ui.showRemainingTasks(tasks);
+                    break;
+                case FIND:
+                    taskInfo = input.split(" ", 2);
+                    ui.showList(tasks.findTasks(taskInfo[1]));
+                    break;
+                case TODO:
+                    taskInfo = input.split(" ", 2);
+                    tempTask = new ToDo(taskInfo[1]);
+                    tasks.addTask(tempTask);
+                    isModified = true;
+                    ui.showAddMessage(tempTask);
+                    ui.showRemainingTasks(tasks);
+                    break;
+                case DEADLINE:
+                    taskInfo = input.split(" ", 2)[1].split(" /by ");
+                    tempTask = new Deadline(taskInfo[0], taskInfo[1]);
+                    tasks.addTask(tempTask);
+                    isModified = true;
+                    ui.showAddMessage(tempTask);
+                    ui.showRemainingTasks(tasks);
+                    break;
+                case EVENT:
+                    taskInfo = input.split(" ", 2)[1].split(" /at ");
+                    tempTask = new Event(taskInfo[0], taskInfo[1]);
+                    tasks.addTask(tempTask);
+                    isModified = true;
+                    ui.showAddMessage(tempTask);
+                    ui.showRemainingTasks(tasks);
+                }
```

Comment:
I like how you made the case clauses in the switch statements to have no indentation, which was highlighted in the Java coding standard. 👍